cdc/sink/dmlsink/factory/factory.go

```diff
@@ -39,14 +39,29 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+// Category is for different DML sink categories.
+type Category = int
+
+const (
+	// CategoryTxn is for Txn sink.
+	CategoryTxn Category = 1
+	// CategoryMQ is for MQ sink.
+	CategoryMQ = 2
+	// CategoryCloudStorage is for CloudStorage sink.
+	CategoryCloudStorage = 3
+	// CategoryBlackhole is for Blackhole sink.
+	CategoryBlackhole = 4
+)
+
 // SinkFactory is the factory of sink.
 // It is responsible for creating sink and closing it.
 // Because there is no way to convert the eventsink.EventSink[*model.RowChangedEvent]
 // to eventsink.EventSink[eventsink.TableEvent].
 // So we have to use this factory to create and store the sink.
 type SinkFactory struct {
-	rowSink dmlsink.EventSink[*model.RowChangedEvent]
-	txnSink dmlsink.EventSink[*model.SingleTableTxn]
+	rowSink  dmlsink.EventSink[*model.RowChangedEvent]
+	txnSink  dmlsink.EventSink[*model.SingleTableTxn]
+	category Category
```

Comment:
The scheme type can be obtained from EventSink, so the category is redunant, ref:

https://github.com/pingcap/tiflow/pull/9688/files#diff-afe94bc54d98b2f898bdce770bba8ee919e3643466203ce4d7087136708e1371R22-R24

Reply:
Different `Schema`s can belong to one same `Category`. Use the latter one can make logic more simple and clear.